chronos/transports/stream.nim

```diff
@@ -699,7 +699,9 @@ when defined(windows):
   proc connect*(address: TransportAddress,
                 bufferSize = DefaultStreamBufferSize,
                 child: StreamTransport = nil,
-                flags: set[TransportFlags] = {}): Future[StreamTransport] =
+                flags: set[TransportFlags] = {},
+                localAddress: TransportAddress = AnyAddress,
+                serverFlags: set[ServerFlags] = {}): Future[StreamTransport] =
```

Comment:
Why we need `ServerFlags` here?

Reply:
To support ReuseAddr & ReusePort.
We could also add ReuseAddr to TransportFlags, but then we get
```
nim-chronos/tests/teststream.nim(349, 61) Error: ambiguous identifier: 'ReuseAddr' -- use one of the following:
  ServerFlags.ReuseAddr: ServerFlags
  TransportFlags.ReuseAddr: TransportFlags
```

so don't see a way to make it backward compatible

Reply:
Its ok, people could use `TransportFlags.ReuseAddr`.

Reply:
Sure, but that breaks everywhere people just use `{ReuseAddr}` instead of `{ServerFlags.ReuseAddr}` (like in these test)
Or we could add `TransportFlags.ClientReuseAddr`

Reply:
So the idea is to avoid double flags in connect() call. So personally i would prefer to introduce `SocketFlags` for both `server` and `client`.

Reply:
Sure, but then I would also need to add `WinServerPipe` & `WinNoPipeFlash` to `SocketFlags`, would that be ok?

Reply:
yep